### PR TITLE
CR-909

### PR DESCRIPTION
**Pull request by andrewjohnysons**
Removed Pending methods and and now only throws PendingException for tests where there is no product available in the Product Service.

Split the SMS tests into normal and Pending tests.

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/fulfilments/TestFulfilmentsEndpoints.java [hunk 1]
```diff
@@ -39,14 +39,7 @@
...
 import uk.gov.ons.ctp.integration.common.product.model.Product;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressDTO;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressQueryResponseDTO;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseDTO;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseType;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.FulfilmentDTO;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.PostalFulfilmentRequestDTO;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.Region;
-import uk.gov.ons.ctp.integration.contactcentresvc.representation.ResponseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.*;
```
Comment (cookel2) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/fulfilments/TestFulfilmentsEndpoints.java:42]: I prefer not to use the .* notation and it hides the information about what package each class has been imported from.

Comment (andrewjohnysons) [reply src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/fulfilments/TestFulfilmentsEndpoints.java:42]: Intellij does that automatically - didn't realise - it's a pain in the arse

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 2]
```diff
@@ -59,13 +59,13 @@ Feature: Test Contact centre Fulfilments Endpoints
   #Scenario CR-T292 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"'
   #because the product reference library does not currently contain the required product - Ella Cook, 27/03/20
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:60]: Please delete this comment about CR-T292 as it no longer applies.

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 3]
```diff
@@ -79,49 +79,49 @@ Feature: Test Contact centre Fulfilments Endpoints
   #Scenario CR-T302 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"'
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:79]: Although this comment about CR-T302 still applies, please delete the comment and instead move the scenario somewhere beneath the new comment that reads #### the following are still PENDING due to lack of products covering these options #####

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 4]
```diff
@@ -79,49 +79,49 @@ Feature: Test Contact centre Fulfilments Endpoints
...
     Then the Case endpoint returns a case, associated with UPRN "1347459993", which has caseType "CE"
     Given a list of available fulfilment product codes is presented for a caseType = "CE" where individual flag = "true" and region = "W"
     And an empty queue exists for sending Fulfilment Requested events
-    When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST" "PENDING" "1347459993"
-    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "true" and region = "W" "PENDING"
+    When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"
+    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "true" and region = "W"
 
   #Scenario CR-T304 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "QUESTIONNAIRE", deliveryChannel "POST"'
   #because the product reference library does not currently contain the required product - Ella Cook, 27/03/20
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:91]: Although this comment about CR-T304 still applies, please delete the comment and instead move the scenario somewhere beneath the new comment that reads #### the following are still PENDING due to lack of products covering these options #####

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 5]
```diff
@@ -79,49 +79,49 @@ Feature: Test Contact centre Fulfilments Endpoints
...
     Then the Case endpoint returns a case, associated with UPRN "1347459993", which has caseType "CE"
     Given a list of available fulfilment product codes is presented for a caseType = "CE" where individual flag = "false" and region = "W"
     And an empty queue exists for sending Fulfilment Requested events
-    When CC Advisor selects the product code for productGroup "QUESTIONNAIRE", deliveryChannel "POST" "PENDING" "1347459993"
-    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "false" and region = "W" "PENDING"
+    When CC Advisor selects the product code for productGroup "QUESTIONNAIRE", deliveryChannel "POST"
+    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "false" and region = "W"
 
   #Scenario CR-T313 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"'
   #because the product reference library does not currently contain the required product - Ella Cook, 30/03/20
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:102]: Although this comment about CR-T313 still applies, please delete the comment and instead move the scenario somewhere beneath the new comment that reads #### the following are still PENDING due to lack of products covering these options #####

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 6]
```diff
@@ -79,49 +79,49 @@ Feature: Test Contact centre Fulfilments Endpoints
...
     Then the Case endpoint returns a case, associated with UPRN "1347459993", which has caseType "CE"
     Given a list of available fulfilment product codes is presented for a caseType = "CE" where individual flag = "true" and region = "N"
     And an empty queue exists for sending Fulfilment Requested events
-    When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST" "PENDING" "1347459993"
-    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "true" and region = "N" "PENDING"
+    When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"
+    Then a fulfilment request event is emitted to RM for UPRN = "1347459993" addressType = "CE" individual = "true" and region = "N"
 
   #Scenario CR-T316 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"'
   #because the product reference library does not currently contain the required product - Ella Cook, 30/03/20
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:113]: Although this comment about CR-T316 still applies, please delete the comment and instead move the scenario somewhere beneath the new comment that reads #### the following are still PENDING due to lack of products covering these options #####

File: src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature [hunk 7]
```diff
@@ -132,41 +132,46 @@ Feature: Test Contact centre Fulfilments Endpoints
   #Scenario CR-T334 throws a pending exception in the step 'When CC Advisor selects the product code for productGroup "UAC", deliveryChannel "POST"'
   #because the product reference library does not currently contain the required product - Ella Cook, 30/03/2
```
Comment (cookel2) [inline src/test/resources/integrationtests/fulfilments/testFulfilmentsEndpoint.feature:133]: Please delete this comment about CR-T334 as it no longer applies.

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/fulfilments/TestFulfilmentsEndpoints.java (hunk 1 above)
Comment (andrewjohnysons) [reply src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/fulfilments/TestFulfilmentsEndpoints.java]: Fixed that now - there is an option in Code style that says use single import class option but for some reson that hasn't worked here :-(

**Pull request merged**